docs/src/mdx-components.tsx

```diff
@@ -9,25 +9,100 @@ import {
-export const thead = TableHead;
-export const tbody = TableBody;
-export const tr = TableRow;
-export const td = TableCell;
-export const th = TableCell;
-export const pre = Pre;
+export const p = ({ children }: React.HTMLAttributes<HTMLParagraphElement>) => (
+  <Text>{children}</Text>
+);
+export const h2 = ({ children }: React.HTMLAttributes<HTMLHeadingElement>) => (
+  <Heading as="h2" type="title1">
+    {children}
+  </Heading>
+);
+export const h3 = ({ children }: React.HTMLAttributes<HTMLHeadingElement>) => (
+  <Heading as="h3" type="title2">
+    {children}
+  </Heading>
+);
+export const h4 = ({ children }: React.HTMLAttributes<HTMLHeadingElement>) => (
+  <Heading as="h4" type="title3">
+    {children}
+  </Heading>
+);
+export const h5 = ({ children }: React.HTMLAttributes<HTMLHeadingElement>) => (
+  <Heading as="h5" type="title4">
+    {children}
+  </Heading>
+);
+export const h6 = ({ children }: React.HTMLAttributes<HTMLHeadingElement>) => (
+  <Heading as="h6" type="title5">
+    {children}
+  </Heading>
+);
+export const table = ({ children }: React.TableHTMLAttributes<HTMLTableElement>) => (
+  <Table>{children}</Table>
+);
+export const thead = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
+  <TableHead>{children}</TableHead>
+);
+export const tbody = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
+  <TableBody>{children}</TableBody>
+);
+export const tr = ({ children }: React.HTMLAttributes<HTMLTableRowElement>) => (
+  <TableRow>{children}</TableRow>
+);
+export const td = ({
+  children,
+  align,
+  valign,
+}: React.TdHTMLAttributes<HTMLTableDataCellElement>) => (
+  <TableCell
+    as="td"
+    align={align === "left" || align === "center" || align === "right" ? align : undefined}
+    verticalAlign={valign}
+  >
+    {children}
+  </TableCell>
+);
+export const th = ({ children, align }: React.ThHTMLAttributes<HTMLTableHeaderCellElement>) => (
+  <TableCell
+    as="th"
+    align={align === "left" || align === "center" || align === "right" ? align : undefined}
+  >
+    {children}
+  </TableCell>
+);
+export const code = CodeBlock;
 export const inlineCode = InlineCode;
-export const a = TextLink;
+export const a = ({ children, href }: React.AnchorHTMLAttributes<HTMLAnchorElement>) => {
+  const isExternal = useIsUrlExternal(href);
+  if (!href) return <a href={href}>{children}</a>;
+  if (!isExternal) return <Link to={href}>{children}</Link>;
+  return (
+    <span
+      css={css`
+        /* TextLink's line-height affects nested elements like <code> */
+        a {
+          line-height: normal;
+        }
+      `}
+    >
+      <TextLink href={href} external>
+        {children}
+      </TextLink>
+    </span>
+  );
+};
+
+function useIsUrlExternal(url: string | void): boolean {
+  return React.useMemo(() => {
+    if (!url) return false;
+    // https://stackoverflow.com/a/2911045/1247274
+    // we can't use URL for this because IE doesn't support it
```

Comment:
should we care about IE at all? i think we agreed, that will not support IE :) 

and it might be outdated, since the link is from 2010 and update there is in 2016 🤔

Reply:
Good question, I guess I didn't remember us declaring that we won't care about IE. In that case I'll use `URL`. And as for the solution possibly being outdated, I'm not _too_ worried about it because it appears to work fine, but I'll write tests for it to confirm.

Reply:
I tried using `URL`, but I realized that I don't really understand how it works, so I sticked to using the anchor element method. I added tests for it to prove that it works well.